### PR TITLE
[Mobile]Update string concatenation for accessibility labels

### PR DESCRIPTION
**Pull request by pinarol**
## Description
Improves some accessibility labels and adds comments to make the localization process easier, based on the feed back at https://github.com/WordPress/gutenberg/pull/15161

## How has this been tested?
Tested via [gutenberg mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/931).

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Review by etoledom — COMMENTED** [review]
Working good!

I found a small detail on Image Block, described on code comments. The rest looks shinny ✨ 

**Another small detail not related to labels:**

The Image block is marked as a 'button' to show that it is actionable. Other blocks are not marked as a button but it says 'actions available'.

In general, I personally prefer mark everything that is actionable with double tap, that doesn't have a more specific trait (like `link`), as button. If what the button action does is not obvious, there we can use hints.

File: packages/block-library/src/image/edit.native.js [hunk 1]
```diff
@@ -395,7 +401,16 @@ class ImageEdit extends React.Component {
...
 							style={ { padding: 12, flex: 1 } }
 							accessible={ true }
-							accessibilityLabel={ __( 'Image caption' ) + __( '.' ) + ' ' + ( isEmpty( caption ) ? __( 'Empty' ) : caption ) }
+							accessibilityLabel={
+								isEmpty( caption ) ?
+									/* translators: accessibility text. Empty image caption. */
+									__( 'Image caption. Empty.' ) :
+									sprintf(
+										/* translators: accessibility text. %s: image caption. */
+										__( 'Image caption. Empty. %s' ),
```
[the remark this answers is not in the document]
Comment (pinarol) [reply packages/block-library/src/image/edit.native.js]: copy paste error :(

**Comment by pinarol** [comment]
Ready for another look @etoledom

**Comment by pinarol** [comment]
> The Image block is marked as a 'button' to show that it is actionable. Other blocks are not marked as a button but it says 'actions available'.

I did some experiments on when the system says 'actions available', It looks like it is related with child components of the block. When there are some actionable components like UITextField inside the block, the system decides saying 'actions available'. For Image block for example, if there's a caption it says 'actions available' but if there's not then it is not saying anything since we aren't rendering the text field. So it looks like we can still benefit from having the button role to indicate available actions.

**Review by etoledom — APPROVED** [review]
Thank you for tiding this up, and for the investigation!

> So it looks like we can still benefit from having the button role to indicate available actions.

I agree on keeping this as it is for now 👍

**Pull request merged**